### PR TITLE
Fix Close error affected by closed nextConn

### PR DESCRIPTION
**Pull request by at-wat**
Stored error in Conn was overwritten by inboundLoop() after Close()
set ErrConnClosed. It caused EOF error of Close() against healthy
connection especially on CI environment.

Investigation of this problem was on https://github.com/pion/webrtc/pull/943.

### CODE REVIEW

File: conn_test.go [hunk 1]
```diff
@@ -73,6 +74,46 @@ func stressDuplex(t *testing.T) {
 	}
 }
 
+func TestStressClose(t *testing.T) {
```
Comment (at-wat) [inline conn_test.go]: This test can reproduce original problem not only on Travis but local.
It doesn't detect in 100%, so `go test . -count=N` would help.

**Comment by Sean-Der** [comment]
@at-wat I am so happy you figured out that issue, it has been driving me crazy for months! I have been playing with your fix and had some questions/thoughts https://github.com/pion/dtls/tree/fix-close-retval-2

#### Add a test that can reproduce this every time
I can do this! Basically make it so you can `Close()` both sides and block the `Read()` and only have it return when we are ready. Then you can call `Close` and `Write` to get the error.

#### Store the first `stopWithErr` call
Right now each call to `stopWithError` will overwrite the previous values, I think we should take the first value only. So if we close with `ErrConnClosed` all the other calls too `stopWithErr` should be discarded.

**Comment by at-wat** [comment]
@Sean-Der 

> Add a test that can reproduce this every time

Yeah, that's way better.
My version of the test consumes long to detect the problem in high probability.

> Store the first `stopWithErr` call

Sounds good!
Returning the first error would be more helpful for the library user as the first error can be assumed as a root of the problem.

**Comment by Sean-Der** [comment]
@at-wat great! I can do all this work and send you a PR (if you don't mind waiting one more day)

**Comment by Sean-Der** [comment]
Moved to #162

**Pull request closed without merging**